Content.Shared/Vehicle/Components/InVehicleComponent.cs

```diff
@@ -0,0 +1,16 @@
+namespace Content.Shared.Vehicle.Components
+{
+    /// <summary>
+    /// Added to objects inside a vehicle to stop people besides the rider from
+    /// removing them.
+    /// </summary>
+    [RegisterComponent]
+    public sealed class InVehicleComponent : Component
+    {
+        /// <summary>
+        /// The vehicle this rider is currently riding.
+        /// </summary>
+        [ViewVariables]
+        public VehicleComponent? Vehicle;
```

Comment:
storing the component instead of uid is potentially bug-prone (since the comp can be deleted and the ref will still exist) but i cant think of any edge cases where it would actually cause bugs

also does it need to be nullable (same for rider)?